File: SDKs/noyo.com/1.0/typescript/src/sdk/models/shared/paginatedcoverageplanresult.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { CoveragePlanResult } from "./coverageplanresult";



// PaginatedCoveragePlanResultMeta
/** 
 * Metadata associated with the response data
**/
export class PaginatedCoveragePlanResultMeta extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=offset" })
  offset: number;

  @SpeakeasyMetadata({ data: "json, name=page_num" })
  pageNum: number;

  @SpeakeasyMetadata({ data: "json, name=page_size" })
  pageSize: number;

  @SpeakeasyMetadata({ data: "json, name=total_records" })
  totalRecords: number;
}


export class PaginatedCoveragePlanResult extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=meta" })
  meta: PaginatedCoveragePlanResultMeta;

  @SpeakeasyMetadata({ data: "json, name=response", elemType: CoveragePlanResult })
  response: CoveragePlanResult[];
}

File: SDKs/noyo.com/1.0/go/pkg/models/shared/paginatedcoverageplanresult.go
package shared

// PaginatedCoveragePlanResultMeta
// Metadata associated with the response data
type PaginatedCoveragePlanResultMeta struct {
	Offset       int32 `json:"offset"`
	PageNum      int32 `json:"page_num"`
	PageSize     int32 `json:"page_size"`
	TotalRecords int32 `json:"total_records"`
}

type PaginatedCoveragePlanResult struct {
	Meta     PaginatedCoveragePlanResultMeta `json:"meta"`
	Response []CoveragePlanResult            `json:"response"`
}
